protocols/virtualfund/virtualfund.go

```diff
@@ -699,15 +698,21 @@ func (o *Objective) updateLedgerWithGuarantee(ledgerConnection Connection, sk *[
 	return sideEffects, nil
 }
 
-// ObjectiveRequest represents a request to create a new virtual funding objective.
-type ObjectiveRequest struct {
+// ObjectiveRequestForVirtualPaymentApp represents a request to create a new virtual funding objective.
+// There is no AppDefinition, since we currently only support full consensus rules for direct channels.
+// There is no Nonce, since the client fills this in.
+type ObjectiveRequestForVirtualPaymentApp struct {
```

Comment:
It seems slightly strange to have a `ObjectiveRequestForVirtualPaymentApp` in `virtualfund` when that struct is only used in the `client` package.

Maybe `ObjectiveRequestForVirtualPaymentApp`(and other structs exposed by API) should live in the `client` package instead of `virtualfund`?

Reply:
We could also just update the API to accept parameters instead of a request struct: `func (c *Client) CreateVirtualChannel(counterparty, intermediary types.Address ....`

Reply:
I ended up going with the parameters approach you suggested (also for the ledger channel API method). Saved a lot of lines! [5284054](https://github.com/statechannels/go-nitro/pull/881/commits/528405467e497575796b3780399ec5398f3b60cc) and [3b340dc](https://github.com/statechannels/go-nitro/pull/881/commits/3b340dc4da702638bc786f38785b80ad0375d803)